### PR TITLE
Add compatibitily matrix

### PR DESCRIPTION
**Pull request by kneth**
## ☑️ ToDos

* [ ] 📝 Changelog entry
* [ ] 📝 `Compatibility` label is updated or copied from previous entry
* [ ] 🚦 Tests
* [ ] 🔀 Executed flexible sync tests locally if modifying flexible sync
* [ ] 📦 Updated internal package version in consuming `package.json`s (if updating internal packages)
* [ ] 📱 Check the React Native/other sample apps work if necessary
* [ ] 📝 Public documentation PR created or is not necessary
* [ ] 💥 `Breaking` label has been applied or is not necessary

*If this PR adds or changes public API's:*
* [ ] typescript definitions file is updated
* [ ] jsdoc files updated
* [ ] Chrome debug API is updated if API is available on React Native

### CODE REVIEW

File: COMPATIBILITY.md [hunk 1]
```diff
@@ -0,0 +1,17 @@
+# React Native and Expo compatibility
+
+
+| Realm JavaScript       | React Native     | Expo         | npm    | node   |
+|------------------------|------------------|--------------|--------|--------|
+| 11.0.0-rc.2            | 0.70             |              |        |        |
```
Comment (tomduncalf) [inline COMPATIBILITY.md:6]: Is this correct, or does rc2 still support >= 0.66.0? The reason I ask I don't think we added any 0.70-specific APIs, but maybe the ABI changed?

File: README.md [hunk 2]
```diff
@@ -15,7 +15,7 @@ This project hosts the JavaScript versions of [Realm](https://realm.io/). Curren
 
 ## Getting Started
 
-Please see the detailed instructions in our docs to use [Realm JavaScript for Node.js](https://docs.mongodb.com/realm/sdk/node/) and [Realm JavaScript for React Native](https://docs.mongodb.com/realm/sdk/react-native/). Please notice that currently only Node.js version 10 or later (excluding 11) are supported.
+Please see the detailed instructions in our docs to use [Realm JavaScript for Node.js](https://docs.mongodb.com/realm/sdk/node/) and [Realm JavaScript for React Native](https://docs.mongodb.com/realm/sdk/react-native/). Please notice that currently only Node.js version 13 or later are supported. Moreover, you might want to check out our [compatibility matrix](COMPATIBILITY.md).
```
Comment (tomduncalf) [inline README.md:18]:
```suggestion
Please see the detailed instructions in our docs to use [Realm JavaScript for Node.js](https://docs.mongodb.com/realm/sdk/node/) and [Realm JavaScript for React Native](https://docs.mongodb.com/realm/sdk/react-native/). Please notice that currently only Node.js version 13 is later are supported. For React Native users, we have a [compatibility matrix](COMPATIBILITY.md) showing which versions are supported.
```

**Review by tomduncalf — APPROVED** [review]

File: COMPATIBILITY.md (hunk 1 above)
Comment (kneth) [reply COMPATIBILITY.md:6]: Good point. As rc2 isn't released yet, I might have made wrong assumption.

Comment (kneth) [reply COMPATIBILITY.md:6]: I will update once rc2 is released.

**Review by takameyer — APPROVED** [review]
Great!  We should share this with the docs team once it's merged

File: COMPATIBILITY.md [hunk 3]
```diff
@@ -0,0 +1,17 @@
+# React Native and Expo compatibility
+
+
+| Realm JavaScript       | React Native     | Expo         | npm    | node   |
```
Comment (kraenhansen) [inline COMPATIBILITY.md:4]: Should we add a JSC / Hermes checkbox column?

File: COMPATIBILITY.md [hunk 4]
```diff
@@ -0,0 +1,18 @@
+# React Native and Expo compatibility
+
+
+| Realm JavaScript       | React Native     | Expo     | Hermes | npm    | node   |
+|------------------------|------------------|----------|--------|--------|--------|
+| 11.0.0-rc.2            | 0.70             | 46       | [x]    | >= 7   | >= 13  |
```
Comment (kraenhansen) [inline COMPATIBILITY.md:6]: According to https://blog.expo.dev/expo-sdk-46-beta-is-now-available-9dfee4040aa7 the 46 release of Expo includes only 0.69.1 and I don't see 0.70.0 mentioned in [their changelog](https://github.com/expo/expo/blob/main/CHANGELOG.md) either.

**Pull request merged**